### ISSUE TITLE
Stop using illegal reflective access

### ISSUE THREAD

**Issue by mustafashabib** [Cosium/spring-data-jpa-entity-graph#23]
Hi All

We're on gradle 5.1 and are seeing this message during builds using spring-data-jpa-entity-graph v2.07.

~~~shell

WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.cosium.spring.data.jpa.entity.graph.repository.support.EntityGraphJpaRepositoryFactory (file:[TRUNCATED]/caches/modules-2/files-2.1/com.cosium.spring.data/spring-data-jpa-entity-graph/2.0.7/397ec4ad1ccef66f9ac5aab08034d837dd3b1067/spring-data-jpa-entity-graph-2.0.7.jar) to field java.lang.reflect.Field.modifiers
WARNING: Please consider reporting this to the maintainers of com.cosium.spring.data.jpa.entity.graph.repository.support.EntityGraphJpaRepositoryFactory
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

~~~

Just wanted to report it in case it's useful.

**Comment by mvmn** [comment]
BTW - some of the tests seem to be failing for your PR there, and also it seems to have merge conflicts now.

**Comment by reda-alaoui** [comment]
Yes I saw that, I will fix it asap ;)

**Comment by reda-alaoui** [comment]
The reference to the Spring Data JPA PR => https://github.com/spring-projects/spring-data-jpa/pull/305

**Comment by reda-alaoui** [comment]
Merge conflict fixed.
Waiting to see if the build turn to green.

**Comment by reda-alaoui** [comment]
Looks like they removed the build step from their PRs.
Just have to wait for review now.

**Comment by jplumez** [comment]
Hello @reda-alaoui 

Any update about the rejected PR ? Will you try to convince them to accept it, or is there any workaround (except disabling illegal access) ?

I have been using your library for a few months and I really like it, it fits my needs perfectly :-) but if the support will stop with Java 12, I should unfortunately reconsider it for some of our projects :( 

Thanks in advance !

**Comment by reda-alaoui** [comment]
Hello @jplumez ,

I am planning to retry to convince them.
But I think the more we are, greater are the chances to convince them.
So I hope you will let a comment on the ticket.

**Comment by reda-alaoui** [comment]
FYI, meanwhile, I think you can still permit illegal access with `--illegal-access=permit`, even with JDK 12.

**Comment by idkw** [comment]
I have commented on the spring issue to add more support from users :)

**Comment by reda-alaoui** [comment]
https://github.com/spring-projects/spring-data-jpa/pull/305 was merged ! :champagne: 

The first Spring Data JPA version to contain https://github.com/spring-projects/spring-data-jpa/pull/305 fix will be 2.3.0. Therefore the current issue fix will be available starting with spring-data-jpa-entity-graph 2.3.0.

**Comment by reda-alaoui** [comment]
Spring Data JPA 2.3.0.RELEASE is now available. 
You can now fix the issue by:
- upgrading to Spring Data JPA 2.3.0.RELEASE
and
- upgrading to Spring Data JPA EntityGraph 2.3.0
